### PR TITLE
Add Support For monochrome Android Adaptive Icons

### PR DESCRIPTION
**Pull request by ngregory-rbi**
# Why

In Android 13, Android added the ability to include a monochrome adaptive icon that will be themed to the OS theme color.

# How

I added this to the already-included Adaptive Icon config, as Android includes it in their [Adaptive Icon docs](https://developer.android.com/develop/ui/views/launch/icon_design_adaptive#design-adaptive-icons).

# Test Plan

Tests written (mostly copied from other image tests).

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

# Notes

I am unsure why the rest of the files changed when I built again. I only changed two source files.

### CODE REVIEW

**Comment by ngregory-rbi** [comment]
Apologies for the multiple commits, I forgot there was a local option to run the tests.

**Comment by ngregory-rbi** [comment]
PR Bump :)

File: packages/@expo/prebuild-config/src/plugins/icons/withAndroidIcons.ts [hunk 1]
```diff
@@ -177,8 +183,15 @@ export async function configureAdaptiveIconAsync(
   projectRoot: string,
   foregroundImage: string,
   backgroundImage: string | null,
+  monochromeImage: string | null,
   isAdaptive: boolean
 ) {
+  if (monochromeImage)
```
Comment (EvanBacon) [inline packages/@expo/prebuild-config/src/plugins/icons/withAndroidIcons.ts:189]: Wrap with curly brackets

Comment (ngregory-rbi) [reply packages/@expo/prebuild-config/src/plugins/icons/withAndroidIcons.ts]: fixed as of [5570270](https://github.com/expo/expo/pull/20074/commits/55702708369cfc4d377a3b29b49c943c9e357a6e) (build commit coming after)

**Review by EvanBacon — CHANGES_REQUESTED** [review]
- fix the build files
- either assert when `foregroundImage` is not defined and `monochromeImage` is defined, or automatically set `foregroundImage` to `monochromeImage` when `foregroundImage` is defined.

File: packages/@expo/config-types/src/ExpoConfig.ts [hunk 2]
```diff
@@ -536,13 +536,17 @@ export interface Android {
      */
     foregroundImage?: string;
     /**
-     * Local path or remote URL to a background image for your app's Adaptive Icon on Android. If specified, this overrides the `backgroundColor` key. Must have the same dimensions as  foregroundImage`, and has no effect if `foregroundImage` is not specified. Should follow the [specified guidelines](https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive).
+     * Local path or remote URL to a background image for your app's Adaptive Icon on Android. If specified, this overrides the `backgroundColor` key. Must have the same dimensions as foregroundImage`, and has no effect if `foregroundImage` is not specified. Should follow the [specified guidelines](https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive).
```
Comment (Simek) [inline packages/@expo/config-types/src/ExpoConfig.ts:539]:
```suggestion
     * Local path or remote URL to a background image for your app's Adaptive Icon on Android. If specified, this overrides the `backgroundColor` key. Must have the same dimensions as `foregroundImage`, and has no effect if `foregroundImage` is not specified. Should follow the [specified guidelines](https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive).
```

Formatting nit.

File: packages/@expo/config-types/src/ExpoConfig.ts [hunk 3]
```diff
@@ -536,13 +536,17 @@ export interface Android {
...
      */
     backgroundImage?: string;
     /**
      * Color to use as the background for your app's Adaptive Icon on Android. Defaults to white, `#FFFFFF`. Has no effect if `foregroundImage` is not specified.
      */
     backgroundColor?: string;
+    /**
+     * (Android 13+ only) Local path or remote URL to an image to use for your app's monochrome Adaptive Icon on Android. Must have the same dimensions as foregroundImage`, and has no effect if `foregroundImage` is not specified. Should follow the [specified guidelines](https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive). This icon will appear on the home screen when the user enables "Themed icons" in system settings.
```
Comment (Simek) [inline packages/@expo/config-types/src/ExpoConfig.ts:547]:
```suggestion
     * (Android 13+ only) Local path or remote URL to an image to use for your app's monochrome Adaptive Icon on Android. Must have the same dimensions as `foregroundImage`, and has no effect if `foregroundImage` is not specified. Should follow the [specified guidelines](https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive). This icon will appear on the home screen when the user enables "Themed icons" in system settings.
```

One more.

**Comment by ngregory-rbi** [comment]
> * fix the build files
> * either assert when `foregroundImage` is not defined and `monochromeImage` is defined, or automatically set `foregroundImage` to `monochromeImage` when `foregroundImage` is defined.

Secondly (sorry for making this a second comment), I am unclear on the second item. Breaking it out into two parts:

1. assert when `foregroundImage` is not defined and `monochromeImage` is defined, or
2. automatically set `foregroundImage` to `monochromeImage` when `foregroundImage` is defined.

For 1, I think that is an impossibility (as `foregroundImage` will default to the `icon` anyway here: https://github.com/expo/expo/blob/main/packages/%40expo/prebuild-config/src/plugins/icons/withAndroidIcons.ts#L40).
For 2, why would I set `foregroundImage` to something else if it's already defined? Was that a typo?

**Comment by ngregory-rbi** [comment]
Weird. The SDK check failed on another file I never changed. Also, sorry for the re-requests. The GitHub UI seems to only let me request one at a time?

**Review by EvanBacon — APPROVED** [review]
fix CI (looks like a linting issue) and we can land. I'll need to update the schema in config-types after the closed source schema change lands.

**Comment by ngregory-rbi** [comment]
> fix CI (looks like a linting issue) and we can land. I'll need to update the schema in config-types after the closed source schema change lands.

It's complaining about a file I never updated, so I'm not sure why it's doing that. Plus, the lines it complains about don't appear to actually exist as it says?

**Comment by ngregory-rbi** [comment]
Made a merge commit on my fork. Hopefully it will fix CI :)

**Comment by ngregory-rbi** [comment]
Also, just wondering, what did you mean by "the closed source schema change"?

**Comment by ngregory-rbi** [comment]
@EvanBacon all checks appear to be good now :)

**Pull request merged**